Plugins/NativeXr/Source/NativeXr.cpp

```diff
@@ -609,68 +623,99 @@ namespace Babylon
 
         void NativeXr::Impl::BeginUpdate()
         {
-            m_sessionState->ActiveTextures.reserve(m_sessionState->Frame->Views.size());
-            for (const auto& view : m_sessionState->Frame->Views)
+            m_sessionState->ActiveViewConfigurations.resize(m_sessionState->Frame->Views.size());
+            for (uint32_t viewIdx = 0; viewIdx < m_sessionState->Frame->Views.size(); viewIdx++)
             {
-                m_sessionState->ActiveTextures.push_back(view.ColorTexturePointer);
+                const auto& view = m_sessionState->Frame->Views[viewIdx];
+                const auto& it{m_sessionState->TextureToViewConfigurationMap.find(view.ColorTexturePointer)};
 
-                auto it{m_sessionState->TextureToFrameBufferMap.find(view.ColorTexturePointer)};
-                if (it == m_sessionState->TextureToFrameBufferMap.end() || it->second->Width() != view.ColorTextureSize.Width || it->second->Height() != view.ColorTextureSize.Height)
+                if (it == m_sessionState->TextureToViewConfigurationMap.end() || 
+                    it->second.ViewTextureSize.Width != view.ColorTextureSize.Width || 
+                    it->second.ViewTextureSize.Height != view.ColorTextureSize.Height ||
+                    it->second.ViewTextureSize.Depth != view.ColorTextureSize.Depth)
                 {
+                    auto& viewConfig = m_sessionState->TextureToViewConfigurationMap[view.ColorTexturePointer] = {};
+                    m_sessionState->ActiveViewConfigurations[viewIdx] = &viewConfig;
+                    m_sessionState->ViewConfigurationStartViewIdx[&viewConfig] = viewIdx;
+
+                    viewConfig.ColorTexturePointer = view.ColorTexturePointer;
+                    viewConfig.DepthTexturePointer = view.DepthTexturePointer;
+                    viewConfig.ViewTextureSize = view.ColorTextureSize;
+
                     // If a texture width or height is 0, bgfx will assert (can't create 0 sized texture). Asserting here instead of deeper in bgfx rendering.
                     assert(view.ColorTextureSize.Width != 0);
                     assert(view.ColorTextureSize.Height != 0);
+                    assert(view.ColorTextureSize.Depth != 0);
```

Comment:
>Depth [](http://example.com/codeflow?start=49&length=5)

nit: update the above comment or call out why depth can't be zero either?

Reply:
Good catch, looks like bgfx accepts 0 layers